src/scrot.c

```diff
@@ -345,6 +346,24 @@ void scrotDoDelay(void)
     }
 }
 
+static inline long miliToNanoSec(int ms)
+{
+    return ms * 1000L * 1000L;
+}
+
+void scrotWaitFor(int ms)
+{
+    scrotAssert(ms >= 0);
+    struct timespec tmp;
+    clock_gettime(CONTINUOUS_CLOCK, &tmp);
+    tmp.tv_sec  += ms / 1000;
+    tmp.tv_nsec += miliToNanoSec(ms % 1000);
+    bool overflow = tmp.tv_nsec >= miliToNanoSec(1000);
+    tmp.tv_sec  += overflow;
+    tmp.tv_nsec -= overflow ? miliToNanoSec(1000) : 0;
+    scrotWaitUntil(&tmp);
+}
```

Comment:
This reintroduces timer drift. The function needs to keep state, it should have a struct timespec 1st parameter that holds a timestamp, then it adds the 2nd parameter to the 1st and sleeps until that moment. 

But then if you do that you'd be reinventing `clock_nanosleep()` + `timespecadd()` when we could simply use those.

Reply:
I don't think this "reintroduces" anything. If you take a look at the 3 call sites, 2 of them aren't called inside a loop (excluding the `EINTR` loop):

```diff
-            while (nanosleep(&t, &t) < 0 && errno == EINTR);
+            scrotSleepFor(160);

-        while (nanosleep(&t, &t) < 0 && errno == EINTR);
+        scrotSleepFor(80);
```

In these 2 cases, it's *fixing* the unlikely drift in case `EINTR` occurred.

- - -

The 3rd case is called inside a loop, but it was already drifting, so nothing is "reintroduced", but one thing is fixed: now it handles `EINTR`.

```diff
-        struct timespec delay = { 0, 50 * 1000L * 1000L };
         do {
-            nanosleep(&delay, NULL);
+            scrotSleepFor(50);
             ret = XGrabKeyboard(disp, root, False, GrabModeAsync, GrabModeAsync, CurrentTime);
         } while (--attempts > 0 && ret == AlreadyGrabbed);
```

As for keeping state, I can try it, but I suspect it'll just be awkward to use (one other reason I didn't use `scrotWaitUntil` on those "fuzzy sleeps").

Reply:
Pushed a commit, the fact that `clock_gettime` doesn't return a value but rather uses an out-param made it more awkward than it needed to be.

(Also just realized that `clock_now` is probably not a great name if `clock_*` is reserved namespace by `time.h`/POSIX)